### DISCUSSION TITLE
Link to homepage filtered by keyword

### DISCUSSION THREAD

**Discussion by Elberet** [verdaccio/verdaccio#1226] [Ideas]
We're running Verdaccio as an internal NPM registry for development tools. Some of these tools use a plugin pattern where the tool inspects another package's dependencies and loads packages with a given keyword as plugins.

I would like to build a link to our Verdaccio instance which automatically fills in the search field and displays the package list filtered by a keyword. I've found code that supposedly does that here:

https://github.com/verdaccio/verdaccio/blob/12d153344b5a95c0fb5b147528ec6db408aaf3c2/src/webui/router.js#L22

Unfortunately, a link of the form `http://npm.corp.example:8080/#/search/my-tool-plugin` results in a blank page; the usual header and footer are rendered, and there are no errors logged in the console.

I'm using the `verdaccio/verdaccio:3.11.4` docker image with the `verdaccio-auth-gitlab` plugin, my browser is Chrome 72.

**Comment by juanpicado** [comment]
Hi @Elberet , yes, I had this on mind since a while. But let me give you an update. The hash UI router does not exist anymore on Version 4 (https://github.com/verdaccio/verdaccio/pull/1197) and is already being tested and develop with other new features we will deliver soon in the `4.x` branch, see here https://github.com/verdaccio/verdaccio/pull/1195 . So, the hash router is nasty, not friendly for new features and that's is the reason why we got rid of it.

So, that said, I'd recommend you apply tour changes on top or (https://github.com/verdaccio/verdaccio/pull/1195) branch if you want to succeed with this feature. 


cc: @ayusharma @priscilawebdev

**Comment by Elberet** [comment]
Ah, thanks for that tip. I didn't realize that a new major version was already in the works, so I thought this was an undocumented and bugged feature.
This feature would have been nice for me right now, but I'm happy to wait until 4.x reaches RC or beta quality and just update our repo. :wink:

**Comment by juanpicado** [comment]
I'll ping you when we reach the beta in the next months :-)

**Comment by Elberet** [comment]
Thanks! 😄
